pkg/imageengine/buildah/common.go

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildah
+
+import (
+	"context"
+
+	"github.com/sealerio/sealer/pkg/define/options"
+
+	"github.com/containers/buildah"
+
+	"os"
+	"time"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/buildah/pkg/parse"
+	"github.com/containers/common/pkg/umask"
+	is "github.com/containers/image/v5/storage"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage"
+	"github.com/containers/storage/pkg/unshare"
+	"github.com/pkg/errors"
+)
+
+const (
+	maxPullPushRetries = 3
+	pullPushRetryDelay = 2 * time.Second
+)
+
+// TODO find a package to place these flags
+const (
+	OCIManifestDir  = "oci-dir"
+	OCIArchive      = "oci-archive"
+	V2s2ManifestDir = "docker-dir"
+	V2s2Archive     = "docker-archive"
+)
+
+func getStore(configurations *options.EngineGlobalConfigurations) (storage.Store, error) {
+	options, err := storage.DefaultStoreOptions(unshare.IsRootless(), unshare.GetRootlessUID())
+	if err != nil {
+		return nil, err
+	}
+
+	if configurations != nil {
+		if len(configurations.GraphRoot) > 0 {
+			options.GraphRoot = configurations.GraphRoot
+		}
+		if len(configurations.RunRoot) > 0 {
+			options.RunRoot = configurations.RunRoot
+		}
+	}
+
+	// Do not allow to mount a graphdriver that is not vfs if we are creating the userns as part
+	// of the mount command.
+	// Differently, allow the mount if we are already in a userns, as the mount point will still
+	// be accessible once "buildah mount" exits.
+	if os.Geteuid() != 0 && options.GraphDriverName != "vfs" {
+		return nil, errors.Errorf("cannot mount using driver %s in rootless mode. You need to run it in a `buildah unshare` session", options.GraphDriverName)
+	}
+
+	umask.Check()
+
+	store, err := storage.GetStore(options)
+	if store != nil {
+		is.Transport.SetStore(store)
+	}
+	return store, err
+}
+
+func openBuilder(ctx context.Context, store storage.Store, name string) (builder *buildah.Builder, err error) {
+	if name != "" {
```

Comment:
What will happen if `name` is empty string? The code logic seems to implicit when `name` is empty string.

Reply:
Will "return nil, errors.Errorf("error finding build container")"